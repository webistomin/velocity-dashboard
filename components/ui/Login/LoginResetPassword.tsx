import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';
import { required } from 'vuelidate/lib/validators';
import { IUserInterface } from 'common/types/user/user-schema';

import BaseTitle from 'components/base/BaseTitle';
import BaseFormGroup from 'components/base/BaseFormGroup';
import BaseButton from 'components/base/BaseButton';
import { serverUrls } from 'common/urls/serverUrls';

export interface ILoginResetForm {
  password: IUserInterface['password'];
  token: string;
}

@Component({
  name: 'LoginResetPassword',
  validations: {
    resetForm: {
      password: {
        required,
      },
    },
  },
})
export default class LoginResetPassword extends VueComponent<{}> {
  public resetForm: ILoginResetForm = {
    password: '',
    token: '',
  };

  public isLoading: boolean = false;

  public async onSubmit(event: Event) {
    event.preventDefault();

    const validator = this.$v;
    validator.$touch();

    if (!validator.$anyError) {
      this.isLoading = true;

      try {
        const data = this.resetForm;
        data.token = this.$route.params.token;
        await this.$axios.$post(serverUrls.auth.reset, data).then((response) => {
          if (response.success) {
            this.$notify({
              group: 'auth',
              type: 'success',
              title: 'Success',
              text: response.message,
              duration: 3000,
            });
          }
          this.isLoading = false;
        });
      } catch (error) {
        console.log(error);
        const data = error.response.data;
        this.$notify({
          group: 'auth',
          type: 'error',
          title: 'Authentication error',
          text: data.message,
          duration: 3000,
        });
        this.isLoading = false;
      }
    }
  }

  public render(): VNode {
    return (
      <div class='login__holder'>
        <div class='login__heading'>
          <BaseTitle class='login__title' level={3}>
            Reset password
          </BaseTitle>
          <p class='login__desc paragraph paragraph_color_darkgray'>Please, enter your new password</p>
        </div>
        <form class='login__form' onSubmit={this.onSubmit}>
          <div class='login__form-inputs'>
            <BaseFormGroup
              class='login__form-group'
              type='password'
              label='New password'
              id='reset-password'
              placeholder='Enter your new password'
              validator={this.$v.resetForm.password}
              onInput={($event: string) => (this.resetForm.password = $event)}
              value={this.resetForm.password}
              onBlur={this.$v.resetForm.password?.$touch}
            />
          </div>
          <BaseButton class='login__submit' type='submit' isLoading={this.isLoading}>
            Submit new password
          </BaseButton>
        </form>
      </div>
    );
  }
}
