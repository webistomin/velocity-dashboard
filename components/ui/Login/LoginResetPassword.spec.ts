import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuelidate from 'vuelidate';
import LoginResetPassword from './LoginResetPassword';

describe('LoginResetPassword', () => {
  const localVue = createLocalVue();
  localVue.use(Vuelidate);
  const factory = (propsData = {}, options = {}) => {
    return shallowMount(LoginResetPassword, {
      localVue,
      propsData: {
        ...propsData,
      },
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
