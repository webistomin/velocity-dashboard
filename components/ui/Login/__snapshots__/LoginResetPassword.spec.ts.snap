// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginResetPassword Matches snapshot 1`] = `
"<div class=\\"login__holder\\">
  <div class=\\"login__heading\\">
    <vuecomponent-stub level=\\"3\\" class=\\"login__title\\">Reset password</vuecomponent-stub>
    <p class=\\"login__desc paragraph paragraph_color_darkgray\\">Please, enter your new password</p>
  </div>
  <form class=\\"login__form\\">
    <div class=\\"login__form-inputs\\">
      <baseformgroup-stub validator=\\"[object Object]\\" messages=\\"[object Object]\\" label=\\"New password\\" attribute=\\"\\" validatorparams=\\"[object Object]\\" type=\\"password\\" id=\\"reset-password\\" value=\\"\\" placeholder=\\"Enter your new password\\" class=\\"login__form-group\\"></baseformgroup-stub>
    </div>
    <basebutton-stub type=\\"submit\\" icon=\\"\\" theme=\\"default\\" class=\\"login__submit\\">Submit new password</basebutton-stub>
  </form>
</div>"
`;
